holoviews/operation/element.py

```diff
@@ -758,8 +758,12 @@ def _process(self, p, element, ranges={}):
         for d1, d2 in permuted_dims:
             if d1 == d2:
                 if p.diagonal_type is not None:
-                    values = element.dimension_values(d1)
-                    el = p.diagonal_type(values, vdims=[d1])
+                    if p.diagonal_type._1d:
```

Comment:
It makes the distinction between Elements that will automatically fill in the x-coordinates and ones that simply render a distribution, e.g. Scatter/Curve vs Distribution/BoxWhisker.